packages/flutter_tools/lib/src/commands/build_aot.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import '../base/process.dart';
+import '../build_configuration.dart';
+import '../globals.dart';
+import '../runner/flutter_command.dart';
+import '../toolchain.dart';
+import 'run.dart';
+
+const String _kDefaultAotOutputDir = 'build/aot';
+
+class BuildAotCommand extends FlutterCommand {
+  BuildAotCommand() {
+    usesTargetOption();
+    addBuildModeFlags();
+    usesPubOption();
+    argParser.addOption('output-dir', defaultsTo: _kDefaultAotOutputDir);
+  }
+
+  @override
+  final String name = 'aot';
+
+  @override
+  final String description = "Build an ahead-of-time compiled snapshot of your app's Dart code.";
+
+  @override
+  Future<int> runInProject() async {
+    String outputPath = buildAotSnapshot(
```

Comment:
Does this take a while to run? If so, let's do `printStatus('Building snapshot...')` (or similar) here.